sdk/src/test/java/io/horizen/account/transaction/EthereumTransactionRlpStreamCodecTest.java

```diff
@@ -351,4 +352,42 @@ public void checkUnsignedTx() {
         byte[] result = writer.toBytes();
         assertEquals(BytesUtils.toHexString(b), BytesUtils.toHexString(result));
     }
+
+    @Test
+    public void checktr201506052141PYTHON()  {
+        // this test is taken from list gethTestVectorsWrongRlp above, and should fail.
+        // The resulting eth tx is parsed OK and is semantically valid instead
+        byte[] b = BytesUtils.fromHexString("f8718439a6c38b88446cf2b7cba3be25847c4af09494a41e36344e8524318a21a527743b169f3a437b8684153aa6b4808189a0f5e5d736775026020ad30508a301eea73d2b096171e6ba17ac3d170f6863b55c9f5fe34e0580ce02b39acae5844a9da787ac7d1a4d97d6bfc53546ba2bc47880");
+        Reader reader = new VLQByteBufferReader(ByteBuffer.wrap(b));
+        EthereumTransaction ethTx = EthereumTransactionDecoder.decode(reader);
+        try {
+            ethTx.semanticValidity();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+        VLQByteBufferWriter writer = new VLQByteBufferWriter(new ByteArrayBuilder());
+        ethTx.encode(true, writer);
+        byte[] result = writer.toBytes();
+        assertEquals(BytesUtils.toHexString(b), BytesUtils.toHexString(result));
+    }
+
+    @Test
+    public void checkIssueEVM823()  {
+        // The tx has a signature r value whose byte array representation is 31 bytes long (not 32), and this should be supported
+        byte[] b = BytesUtils.fromHexString("f8878214920783011170941050072833849ff9bf49a55ff15d40f6d89651e880a440d097c30000000000000000000000000f6cf5090c089dfbd2c37eb58a70abad83dd79d2820d209f52759db97387354cd02079a382b1a165bac472be1da47a15a748039c6faf18a032bff10a83396a6a4a89b2f18880d6d73f7b6af9952b3f06cacc1283c86f0fa8");
+        Reader reader = new VLQByteBufferReader(ByteBuffer.wrap(b));
+        EthereumTransaction ethTx = EthereumTransactionDecoder.decode(reader);
+        try {
+            ethTx.semanticValidity();
```

Comment:
since we don't expect any exception here: let's add a `fail(...)` inside `catch` block

Reply:
Ok, fixed